File: 151001/Yankova/RV2/src/main/java/com/bsuir/nastassiayankova/services/TagService.java
package com.bsuir.nastassiayankova.services;

import com.bsuir.nastassiayankova.beans.entity.ValidationMarker;
import com.bsuir.nastassiayankova.beans.request.TagRequestTo;
import com.bsuir.nastassiayankova.beans.response.TagResponseTo;
import jakarta.validation.Valid;
import org.springframework.validation.annotation.Validated;

import java.util.List;

public interface TagService {
    @Validated(ValidationMarker.OnCreate.class)
    TagResponseTo create(@Valid TagRequestTo entity);

    List<TagResponseTo> read();

    @Validated(ValidationMarker.OnUpdate.class)
    TagResponseTo update(@Valid TagRequestTo entity);

    void delete(Long id);

    TagResponseTo findTagById(Long Id);
}

File: 151004/Bashlikov/rv_task01/src/main/kotlin/by/bashlikovvv/services/TagService.kt
package by.bashlikovvv.services

import by.bashlikovvv.api.dto.request.CreateTagDto
import by.bashlikovvv.api.dto.request.UpdateTagDto
import by.bashlikovvv.api.dto.response.TagDto

interface TagService {

    fun create(createTagDto: CreateTagDto): TagDto?

    fun getAll(): List<TagDto?>

    fun getById(tagId: Long): TagDto?

    fun update(tagId: Long, updateTagDto: UpdateTagDto): TagDto?

    fun delete(tagId: Long): Boolean

}